ui/src/pages/frontsite/PatronProfile/PatronPendingLoans/PatronPendingLoans.js

```diff
@@ -154,6 +82,46 @@ export default class PatronPendingLoans extends Component {
     );
   };
 
+  onShowCancelModal = (e, loan) => {
+    this.setState({
+      cancelModal: { isOpen: true, btnClasses: e.target.classList, data: loan },
+    });
+  };
+
+  onCloseCancelModal = () => {
+    this.setState({
+      cancelModal: { isOpen: false, data: undefined },
+    });
+  };
+
+  onCancelRequestClick = () => {
+    const loan = this.state.cancelModal.data;
+    const { btnClasses } = this.state.cancelModal;
+    btnClasses.add('disabled');
```

Comment:
the modal should add the classes to the button, not the parent component. 
in the modal you could have something like 
```
 triggerClickedHandler = () => {
    set_my_disabled_and_loading_classes()
    this.props.onCancelAction()
    setTimeout (() => {
        this.props.cancelCompletedCallback()
    }
}
```
in general the modal should handle its internal business

Reply:
@kprzerwa, I hear you, we should talk about this one a bit more I believe, I am not sure if we should bind them together, the triggers (aka buttons) and the modal.

Reply:
I am not fully understand the code in the `onCancelRequestClick` method, anyway I would go a little bit further of what Karolina suggested.
The modal should not know the "cancel" action, actually I find it confusing and mixed with the "cancel/close" button.
Ideally, the modal interface/contract should be simply:
```
Modal(title, content, onBtnDismissClicked, onBtnSubmitClicked)
```
where dismiss is probably triggered when clicking on cancel button but also when closing the modal with the top right `X`, if exists.
Then, it is up to the caller to implement the callbacks:
```
Caller

new Modal("my title", "my content", function () {alert("cancel clicked"), function() {alert("submit clicked")} )}
```

Maybe it could have an extra option for closing it or not automatically...
IMHO, the modal should wrap the click handlers by disabling the buttons first:
```
Modal {
...
    _onBtnSubmitClicked = () => {
        disable_this_button();
        this.onBtnSubmitClicked()
        this.close();
    }
...
}
```

Reply:
can you please fix this as discussed in one single new PR? @kprzerwa hopefully you agree.